Fix overlong junit filename prefixes

ivan4th opened the pull request:
Build scripts construct filenames that are longer than 255 chars when producing JUnit output for tests and `KUBE_COVER=y`:

```
vagrant@devbox:~/work/kubernetes/src/k8s.io/kubernetes (test %) $ KUBE_JUNIT_REPORT_DIR=/tmp/art KUBE_COVER=y make test
E0818 15:33:29.312549    9426 conversion.go:589] Warning: could not generate autoConvert functions for k8s.io/kubernetes/pkg/apis/extensions/v1beta1.HorizontalPodAutoscalerSpec <-> k8s.io/kubernetes/pkg/apis/autoscaling.HorizontalPodAutoscalerSpec
E0818 15:33:29.323685    9426 conversion.go:589] Warning: could not generate autoConvert functions for k8s.io/kubernetes/pkg/apis/extensions/v1beta1.JobSpec <-> k8s.io/kubernetes/pkg/apis/batch.JobSpec
E0818 15:33:29.344775    9426 conversion.go:589] Warning: could not generate autoConvert functions for k8s.io/kubernetes/pkg/apis/extensions/v1beta1.RollingUpdateDeployment <-> k8s.io/kubernetes/pkg/apis/extensions.RollingUpdateDeployment
E0818 15:33:29.349260    9426 conversion.go:589] Warning: could not generate autoConvert functions for k8s.io/kubernetes/pkg/apis/extensions/v1beta1.ScaleStatus <-> k8s.io/kubernetes/pkg/apis/extensions.ScaleStatus
Running tests for APIVersion: v1,apps/v1alpha1,authentication.k8s.io/v1beta1,authorization.k8s.io/v1beta1,autoscaling/v1,batch/v1,batch/v2alpha1,certificates/v1alpha1,extensions/v1beta1,federation/v1beta1,policy/v1alpha1,rbac.authorization.k8s.io/v1alpha1,imagepolicy.k8s.io/v1alpha1
+++ [0818 15:33:30] Saving coverage output in '/tmp/k8s_coverage/v1,apps/v1alpha1,authentication.k8s.io/v1beta1,authorization.k8s.io/v1beta1,autoscaling/v1,batch/v1,batch/v2alpha1,certificates/v1alpha1,extensions/v1beta1,federation/v1beta1,policy/v1alpha1,rbac.authorization.k8s.io/v1alpha1,imagepolicy.k8s.io/v1alpha1/20160818-153330'
tee: /tmp/art/junit_v1,apps-v1alpha1,authentication.v1beta1,authorization.v1beta1,autoscaling-v1,batch-v1,batch-v2alpha1,certificates-v1alpha1,extensions-v1beta1,federation-v1beta1,policy-v1alpha1,rbac.authorization.v1alpha1,imagepolicy.v1alpha1_20160818-153330-cmd_genutils.stdout: File name too long
tee: /tmp/art/junit_v1,apps-v1alpha1,authentication.v1beta1,authorization.v1beta1,autoscaling-v1,batch-v1,batch-v2alpha1,certificates-v1alpha1,extensions-v1beta1,federation-v1beta1,policy-v1alpha1,rbac.authorization.v1alpha1,imagepolicy.v1alpha1_20160818-153330-cmd_kube-apiserver_app.stdout: File name too long
tee: /tmp/art/junit_v1,apps-v1alpha1,authentication.v1beta1,authorization.v1beta1,autoscaling-v1,batch-v1,batch-v2alpha1,certificates-v1alpha1,extensions-v1beta1,federation-v1beta1,policy-v1alpha1,rbac.authorization.v1alpha1,imagepolicy.v1alpha1_20160818-153330-cmd_hyperkube.stdout: File name too long
tee: /tmp/art/junit_v1,apps-v1alpha1,authentication.v1beta1,authorization.v1beta1,autoscaling-v1,batch-v1,batch-v2alpha1,certificates-v1alpha1,extensions-v1beta1,federation-v1beta1,policy-v1alpha1,rbac.authorization.v1alpha1,imagepolicy.v1alpha1_20160818-153330-cmd_kube-apiserver_app_options.stdout: File name too long
ok      k8s.io/kubernetes/cmd/genutils  0.002s
tee: /tmp/art/junit_v1,apps-v1alpha1,authentication.v1beta1,authorization.v1beta1,autoscaling-v1,batch-v1,batch-v2alpha1,certificates-v1alpha1,extensions-v1beta1,federation-v1beta1,policy-v1alpha1,rbac.authorization.v1alpha1,imagepolicy.v1alpha1_20160818-153330-cmd_kubelet_app.stdout: File name too long
qok     k8s.io/kubernetes/cmd/kube-apiserver/app/options    0.159s
tee: /tmp/art/junit_v1,apps-v1alpha1,authentication.v1beta1,authorization.v1beta1,autoscaling-v1,batch-v1,batch-v2alpha1,certificates-v1alpha1,extensions-v1beta1,federation-v1beta1,policy-v1alpha1,rbac.authorization.v1alpha1,imagepolicy.v1alpha1_20160818-153330-cmd_kube-proxy_app.stdout: File name too long
```

Let's shorten filenames a bit more to make them fit in 255 chars. This is probably temporary solution too, the bulletproof one being (e.g.) using `sha1sum` of `KUBE_TEST_API` but that would produce files with meaningless filenames so for now maybe it may wait.

(there was recent attempt to shorten the filenames by removing `k8s.io`s, but that was not enough for `KUBE_COVER=y`)

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.kubernetes.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/30894)

<!-- Reviewable:end -->

Commit:
Fix overlong junit filename prefixes

lavalamp commented on hack/make-rules/test.sh:
```diff
@@ -157,7 +157,10 @@ junitFilenamePrefix() {
   local KUBE_TEST_API_NO_SLASH="${KUBE_TEST_API//\//-}"
   # This file name isn't parsed by anything, and tee needs a shorter file name.
+  # Convert strings like junit_v1,apps-v1alpha1,authentication.k8s.io-v1beta1
+  # to junit_v1,apps.k-v1a1,authentication.k-v1b1
```
The most important part of a comment is the why, not the what! Please tell future us why you're making this change. :)

lavalamp commented:
Please fix comment and we'll get this in asap.

ivan4th commented:
*[hack/make-rules/test.sh, line 161 \[r1\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/30894#-KPTQMKtEPAra6QVTXQL-r1-161:-KPUDoNBMAXD8m4tapW_:bv0f0hr) ([raw file](https://github.com/kubernetes/kubernetes/blob/79058a19b03ddc80d4e7eaa4d4316d24ed7c7a8d/hack/make-rules/test.sh#L161)):*
<details><summary><i>Previously, lavalamp (Daniel Smith) wrote…</i></summary>
> The most important part of a comment is the why, not the what! Please tell future us why you're making this change. :)
</details>
There was earlier comment above this explanation concerning the overlong filenames (someone was trying to make filenames shorter but not hard enough :), but it was somewhat wrong because 255 char limit isn't imposed by tee, so I fixed it. As of what the sed command does, I thought that it wasn't very readable (like many sed commands), so I added the description, but ok, perhaps it's indeed a little redundant.

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/30894)*
<!-- Sent from Reviewable.io -->

ivan4th commented:
Fixed.

---







*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/30894#-:-KPUDUgW4pxf3VlKd5H7:bt2vr1b)*
<!-- Sent from Reviewable.io -->

lavalamp commented:
LGTM, thanks!!

lavalamp commented:
In fact, the queue is currently blocked presumably because it can't parse junit from a flaky test-go run.

@krousey (build cop) how do you feel about manually merging this?

krousey commented:
@lavalamp I'm find with a manual merge, but I don't see how this is blocking the build. There's a data race detection, but I don't see the junit parsing error.

krousey commented:
Manual merge on faith that @lavalamp saw something that I couldn't find.

lavalamp commented:
I saw a failed individual test, but queue made it red instead of orange,
which means it failed to find a relevant junit file to parse, presumably
because the upload failed due to the name being huge.

On Thu, Aug 18, 2016 at 3:04 PM, krousey <notifications@github.com> wrote:

> Manual merge on faith that @lavalamp <https://github.com/lavalamp> saw
> something that I couldn't find.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/30894#issuecomment-240871457>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAnglgrr7LtAQZuJGzIWQ0dVi4tFEaj4ks5qhNbrgaJpZM4JnqJw>
> .
>
